trie/database.go

```diff
@@ -61,33 +133,46 @@ func (t *Trie) writeToDB(n node) error {
 	return nil
 }
 
-// writeNodeToDB returns true if node is written to db batch writer, false otherwise
-// if node is clean, it will not attempt to be written to the db
-// otherwise if it's dirty, try to write it to db
-func (t *Trie) writeNodeToDB(n node) (bool, error) {
-	if !n.isDirty() {
-		return false, nil
+// StoreInDB encodes the entire trie and writes it to the DB
+// The key to the DB entry is the root hash of the trie
+func (t *Trie) StoreInDB() error {
+	enc, err := t.Encode()
+	if err != nil {
+		return err
 	}
 
-	encRoot, err := Encode(n)
+	roothash, err := t.Hash()
 	if err != nil {
-		return false, err
+		return err
 	}
 
-	hash, err := t.db.Hasher.Hash(n)
+	return t.db.Db.Put(roothash[:], enc)
+}
+
+// LoadFromDB loads an encoded trie from the DB where the key is `root`
+func (t *Trie) LoadFromDB(root common.Hash) error {
+	enctrie, err := t.db.Db.Get(root[:])
 	if err != nil {
-		return false, err
+		return err
 	}
 
-	t.db.Lock.Lock()
-	err = t.db.Batch.Put(hash[:], encRoot)
-	t.db.Lock.Unlock()
+	return t.Decode(enctrie)
+}
+
+func (t *Trie) StoreHash() error {
+	hash, err := t.Hash()
+	if err != nil {
+		return err
+	}
 
-	n.setDirty(false)
-	return true, err
+	return t.db.Db.Put(LATEST_HASH_KEY, hash[:])
 }
 
-// Commit writes the contents of the db's batch writer to the db
-func (t *Trie) Commit() error {
-	return t.db.Batch.Write()
+func (t *Trie) LoadHash() (common.Hash, error) {
```

Comment:
Please comment. I think `LoadLatestHash` might be less ambiguous

Reply:
done